go/vt/vttablet/tabletmanager/vdiff/controller.go

```diff
@@ -162,13 +161,21 @@ func (ct *controller) updateState(dbClient binlogplayer.DBClient, state VDiffSta
 		extraCols = ", completed_at = utc_timestamp()"
 	default:
 	}
+	var errorString string
 	if err == nil {
 		// Clear out any previous error for the vdiff on this shard
-		err = errors.New("")
+		errorString = ""
+	} else {
+		// limit the error string to be within column length of `last_error`
+		const MaxErrorLength = 500
+		errorString = err.Error()
+		if len(errorString) > MaxErrorLength {
+			errorString = errorString[:MaxErrorLength]
+		}
```

Comment:
I think that we should instead (or in addition) modify the table structure: https://github.com/vitessio/vitess/blob/main/go/vt/sidecardb/schema/vdiff/vdiff.sql

It doesn't make sense to me that e.g. the workflow name and db name fields are twice as long as the last error field. This is important info that we should avoid losing when possible. I think that we should at least make the field as long as the workflow field. We should still check the length though to be safe.

We could also truncate the error in the SQL statement instead: `LEFT("val...", 500)`

Reply:
I'll add a commit for this and my other nits.